app/ota_client_stub.py

```diff
@@ -95,14 +97,21 @@ def start(self, enable_cache=False):
             else:
                 raise Exception("server already started")
 
-    def stop(self):
+    def stop(self, cleanup_cache=False):
+        from ota_proxy.config import config as proxy_srv_cfg
+
         with self._lock:
             if not self._closed:
                 # send SIGTERM to the server process
                 self._server_p.terminate()
                 self._closed = True
                 logger.info("ota proxy server closed")
 
+                if cleanup_cache:
```

Comment:
I think cleanup is not required since this is called just before switching partition.
How do you think?

Reply:
Do you mean I should cleanup the cache after first successful reboot?

Reply:
after successful reboot, ota cache directory is disappeared since the partition has been switched, isn't it?

Reply:
> after successful reboot, ota cache directory is disappeared since the partition has been switched, isn't it?

Yes, so I think the best time to clear cache is the POST_UPDATING stage, before reboot. In this stage, all subECUs are ready and local ota update also finished.